docs/developer/architecture/alerting.rst

```diff
@@ -43,3 +43,295 @@ maintenance operation.
+
+This API call should be done in a Salt module ``metalk8s_monitoring``
+which could then be wrapped in a CLI and UI.
+
+.. _Prometheus API: https://prometheus.io/docs/prometheus/latest/querying/api
+
+Silence an Alert
+~~~~~~~~~~~~~~~~
+
+To silence an alert, we need to send a query to the Alertmanager API.
+This can be done using the following route::
+
+    http://<alertmanager-ip>:9093/api/v1/silences
+
+With a POST query content formatted as below::
+
+    {
+      "matchers": [
+        {
+          "name": "<alert-name>",
+          "value": ".*",
+          "isRegex": true
+        }
+      ],
+      "startsAt": "2020-04-10T12:12:12",
+      "endsAt": "2020-04-10T13:12:12",
+      "createdBy": "<author>",
+      "comment": "Maintenance is planned",
+      "status": {
+        "state": "active"
+      }
+    }
+
+We must also be able to retrieve silenced alerts and to remove a silence.
+This will be done using the API, with the same route using GET and DELETE word
+respectively::
+
+    # GET - to list all silences
+    http://<alertmanager-ip>:9093/api/v1/silences
+
+    # DELETE - to delete a specific silence
+    http://<alertmanager-ip>:9093/api/v1/silence/<silence-id>
+
+We will need to provide these functionnalities through a Salt module
+``metalk8s_monitoring`` which could then be wrapped in a CLI and UI.
+
+Extract Rules Tooling
+---------------------
+
+We need to build a tool to extract all alert rules from the :term:`Prometheus`
+Operator rendered chart
+``salt/metalk8s/addons/prometheus-operator/deployed/chart.sls``.
+
+Its purpose will be to generate a file (each time this chart is updated)
+which will then be used to check that what's deployed matches what was
+expected.
+
+And so, we will be able to see what has been changed when updating
+:term:`Prometheus` Operator chart and see if there is any change on
+customizable alert rules.
+
+Rejected Design Choices
+-----------------------
+
+Using amtool_ vs Alertmanager API
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Managing alert silences can be done using amtool_::
+
+    # Add
+    amtool --alertmanager.url=http://localhost:9093 silence add \
+      alertname="<alert-name>" --comment 'Maintenance is planned'
+
+    # List
+    amtool --alertmanager.url=http://localhost:9093 silence query
+
+    # Delete
+    amtool --alertmanager.url=http://localhost:9093 silence expire <silence-id>
+
+This option has been rejected because, to do so, we need to install an extra
+dependency (amtool_ binary) or run the commands inside the :term:`Alertmanager`
+container, rather than simply send HTTP queries on the API.
+
+.. _amtool: https://github.com/prometheus/alertmanager/blob/master/README.md#amtool
+
+Implementation Details
+----------------------
+
+Iteration 1
+~~~~~~~~~~~
+
+* Add an internal tool to list all Prometheus alert rules from rendered chart
+* Implement Salt formulas to handle configuration customization
+  (``advanced`` mode only)
+* Provide CLI and UI to wrap the Salt calls
```

Comment:
@alexandre-allard-scality @TeddyAndrieux I am not sure I listed those commands in centralised CLI spec, should I add it?

Reply:
I didn't see them. I think we may want to put them in the centralized CLI spec.